src/inmanta/protocol/auth/default_policy.rego

```diff
@@ -0,0 +1,159 @@
+package policy
+
+# Write the information about the endpoint into a variable
+# to make the policy easier to read.
+endpoint_data := data.endpoints[input.request.endpoint_id]
+
+# The environment used in the request
+request_environment := input.request.parameters[endpoint_data.environment_param] if {
+    endpoint_data.environment_param != null
+} else := null
+
+# Don't allow anything that is not explicitly allowed.
+default allowed := false
+
+### General rules ###
+
+# Every user can change its own password
+allowed if {
+    input.request.endpoint_id == "PATCH /api/v2/user/<username>/password"
+    input.token.sub == input.request.parameters.username
+}
+
+# Any authenticated user is allowed read-only access on the environment independent endpoints.
+allowed if {
+    endpoint_data.read_only
+    endpoint_data.environment_param == null
+}
+
+### Role: read-only ###
+
+# Users with the read-only role can execute all read-only endpoints on that environment.
+
+read_only_labels := {
+    "agent.read",
+    "compilereport.read",
+    "compiler.status.read",
+    "desired-state.read",
+    "discovered_resources.read",
+    "docs.read",
+    "dryrun.read",
+    "environment.read",
+    "environment.settings.read",
+    "fact.read",
+    "files.read",
+    "graphql.read",
+    "metrics.read",
+    "notification.read",
+    "parameter.read",
+    "pip_config.read",
+    "project.read",
+    "resources.read",
+    "status.read",
+    "lsm.callback.read",
+    "lsm.catalog.read",
+    "lsm.docs.read",
+    "lsm.instance.read",
+    "lsm.order.read",
+    "support.support-archive.read",
+    "graphql.read",
+}
+
+allowed if {
+    "read-only" in input.token["urn:inmanta:roles"][request_environment]
+    endpoint_data.auth_label in read_only_labels
+}
+
+
+## Role: noc ###
+
+# Can do all operations in an environment that do not alter the desired state or modify the settings.
+
+noc_specific_labels := {
+    "compiler.execute",
```

Comment:
I wonder if "update project & recompile" shouldn't be considered an admin action as well? Admittedly, it's a bit more difficult than the others because it's the same endpoint as plain recompile. But it does "alter the desired state", and in such a way that it more or less allows remote code execution, provided you also have write access to the configured repository.

Reply:
I agree that "update and recompile" allows you to do remote code execution, but noc users need to be able to execute this operation as part of their job. So I don't think we can make it an admin action. Also, I don't think the idea here is to protect against people with bad intentions, but more to only give users the access they actually need. @bartv Do you have an opinion on this?

Reply:
so for me that one has a different level of access than a normal compile. I lean towards admin as well, NOC users really do not do this.